src/main/java/com/capitalone/dashboard/collector/SonarClientSelector.java

```diff
@@ -1,20 +1,63 @@
 package com.capitalone.dashboard.collector;
 
 
+import java.net.URI;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestOperations;
+
+import com.capitalone.dashboard.util.Supplier;
 
 @Component
 public class SonarClientSelector {
+	private static final Log LOG = LogFactory.getLog(SonarClientSelector.class);
+
+	private static final String URL_VERSION_RESOURCE = "/api/server/version";
 
-    @Autowired
     private DefaultSonar6Client sonar6Client;
-    @Autowired
     private DefaultSonar56Client sonar56Client;
-    @Autowired
-    @Qualifier("DefaultSonarClient")
     private DefaultSonarClient sonarClient;
+    private RestOperations rest;
+    
+    @Autowired
+    public SonarClientSelector(
+    		DefaultSonar6Client sonar6Client, DefaultSonar56Client sonar56Client,
+            @Qualifier("DefaultSonarClient") DefaultSonarClient sonarClient,
+            Supplier<RestOperations> restOperationsSupplier) {
+
+        this.sonar6Client = sonar6Client;
+        this.sonar56Client = sonar56Client;
+        this.sonarClient = sonarClient;
+        this.rest = restOperationsSupplier.get();
+    }
+    
+    public Double getSonarVersion(String instanceUrl){
+    	Double version = 6.0;
```

Comment:
can this default version be set to 5 so that its backward compatible, I dont believe the version api will work with sonar 5.x